### PR TITLE
Add sphinx-mermaid extension and update userdoc workflow

### PR DESCRIPTION
**Pull request by jessica-mitchell**
This PR adds the mermaid extension for Sphinx so sequence diagrams can be build with mermaid syntax.
See [kernel.html](https://nest-test.readthedocs.io/en/add-sphinx-mermaid/developer_space/architecture/kernel.html) for example.

### CODE REVIEW

File: doc/htmldoc/conf.py [hunk 1]
```diff
@@ -253,6 +256,7 @@ def setup(app):
     app.add_css_file('css/custom.css')
     app.add_css_file('css/pygments.css')
     app.add_js_file("js/custom.js")
+    app.add_js_file("js/mermaid.js")
```
Comment (steffengraber) [inline doc/htmldoc/conf.py:259]: Is this really necessary? Isn't the JavaScript automatically integrated when the plug-in is activated?

**Review by steffengraber — COMMENTED** [review]
Thank you, this is really a useful extension.

**Review by JanVogelsang — APPROVED** [review]
Steffen's comment sounds plausible (also after taking a quick look at the spinx-mermaid installation guide), apart from that this is a nice addition that I hope we can use a lot to make the usually boring technical documentation much more visual, as an example.

**Review by steffengraber — APPROVED** [review]
:+1: Great!

Do you have a example page?

**Comment by jessica-mitchell** [comment]
See new documentation workflow output [here](https://nest-simulator--2603.org.readthedocs.build/en/2603/developer_space/workflows/documentation_workflow/user_documentation_workflow.html)

**Review by steffengraber — APPROVED** [review]
Looks good!

**Pull request merged**